app/src/main/java/com/nft/signdoc/ui/auth/CreateAccActivity.kt

```diff
@@ -1,28 +1,144 @@
+import android.text.InputType
+import android.text.Spannable
+import android.text.SpannableString
+import android.text.style.ForegroundColorSpan
 import android.view.View
+import android.widget.Toast
+import androidx.activity.viewModels
 import androidx.annotation.LayoutRes
+import androidx.core.content.ContextCompat
+import androidx.core.widget.doAfterTextChanged
 import androidx.fragment.app.Fragment
 import com.nft.signdoc.R
 import com.nft.signdoc.databinding.ActivityCreateAccountBinding
+import com.nft.signdoc.extensions.makeLinks
 import com.nft.signdoc.extensions.viewBinding
 import com.nft.signdoc.ui.base.BaseActivity
 import com.nft.signdoc.ui.base.BaseFragment
+import com.nft.signdoc.util.Helpers
+import com.nft.signdoc.viewmodel.UserViewModel
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
 class CreateAccActivity : BaseActivity() {
 
+
+    companion object {
+        fun getIntent(context: Context): Intent {
+            return Intent(context, CreateAccActivity::class.java)
+        }
+    }
+
     private val binding by viewBinding(ActivityCreateAccountBinding::inflate)
 
+    private val userViewModel : UserViewModel by viewModels()
+
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(binding.root)
+        initViews()
         listenToViewEvents()
     }
 
+    private fun initViews(){
+        binding.termsText.makeLinks(Pair("Terms & Conditions", View.OnClickListener {
+            val browserIntent = Intent(Intent.ACTION_VIEW)
+            browserIntent.data = Uri.parse("https://terms.nftmakerapp.io/")
+            this.startActivity(browserIntent)
+        }), Pair("Privacy Policy", View.OnClickListener {
+            val browserIntent = Intent(Intent.ACTION_VIEW)
+            browserIntent.data = Uri.parse("https://privacy.nftmakerapp.io/")
+            this.startActivity(browserIntent)
+        }))
+
+        binding.loginButtonView.apply {
+
+            val wordtoSpan: Spannable =
+                SpannableString(getString(R.string.already_have_account))
+
+            wordtoSpan.setSpan(
+                ForegroundColorSpan(getColor(R.color.dark_blue)),
+                27,
+                44,
+                Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
+            )
+
+            this.text = wordtoSpan
+        }
+
+    }
+
     private fun listenToViewEvents() {
 
+        binding.emailBtn.setOnClickListener {
+            binding.EmailPhone.text?.clear()
+            binding.emailBtn.backgroundTintList = ColorStateList.valueOf(resources.getColor(R.color.selectedColor))
+            binding.phoneBtn.backgroundTintList = ColorStateList.valueOf(Color.WHITE)
+            binding.EmailPhone.hint = this.getString(R.string.email_example)
+            userViewModel.usesPhone = false
+            binding.EmailPhone.inputType = InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS
+            binding.signUpMethod.text = getString(R.string.email)
+
```

Comment:
When user click on this button his input will be reset either he is entering his mobile number or email. For example user is entering his email and accidentally click on this button, his email will be reset and user have to enter his email again from the start. 
You should disable this button once user has selected it until user click on the phoneBtn option and than click on this button.

Same goes for phoneBtn.

Reply:
actually this behavior is not my decision, its the same behavior as NFT Maker app, ofcourse that decision came from QA and then from upper management so yeah